cmds/core/wget/wget_test.go

```diff
@@ -40,131 +43,147 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-var tests = []struct {
-	name    string
-	flags   []string // in, %[1]d is the server's port, %[2] is an unopen port
-	url     string   // in
-	content string   // out
-	retCode int      // out
-}{
-	{
-		name:    "basic",
-		flags:   []string{},
-		url:     "http://localhost:%[1]d/200",
-		content: content,
-		retCode: 0,
-	},
-	{
-		name:    "ipv4",
-		flags:   []string{},
-		url:     "http://127.0.0.1:%[1]d/200",
-		content: content,
-		retCode: 0,
-	},
-	// TODO: CircleCI does not support ipv6
-	// {
-	// 	name:    "ipv6",
-	// 	flags:   []string{},
-	// 	url:     "http://[::1]:%[1]d/200",
-	// 	content: content,
-	// 	retCode: 0,
-	// },
-	{
-		name:    "redirect",
-		flags:   []string{},
-		url:     "http://localhost:%[1]d/302",
-		content: "",
-		retCode: 0,
-	},
-	{
-		name:    "4xx error",
-		flags:   []string{},
-		url:     "http://localhost:%[1]d/404",
-		content: "",
-		retCode: 1,
-	},
-	{
-		name:    "5xx error",
-		flags:   []string{},
-		url:     "http://localhost:%[1]d/500",
-		content: "",
-		retCode: 1,
-	},
-	{
-		name:    "no server",
-		flags:   []string{},
-		url:     "http://localhost:%[2]d/200",
-		content: "",
-		retCode: 1,
-	},
-}
-
-func getListener(t *testing.T) (net.Listener, int) {
+// unusedPort return a port number and setup a listener that closes connection
+// for any request
+func unusedPort(t *testing.T) (int, func() error) {
```

Comment:
Now that the code is actually listening and not returning an "unused port" - can you think of a better name & docstring for this function? Thanks!

Reply:
updated